packages/router/test/integration.spec.ts

```diff
@@ -76,6 +76,47 @@ describe('Integration', () => {
          ]);
        })));
 
+    // https://github.com/angular/angular/issues/26983
+    describe('relativeLinkResolution', () => {
+      beforeEach(inject([Router], (router: Router) => {
+        router.resetConfig([
+          {
+            path: 'foo/bar',
+            children: [
+              {
+                path: '',
+                component: RelativeLinkCmp
+              }
+            ]
+          },
+        ]);
+      }));
+
+      it('should not ignore empty paths in legacy mode', fakeAsync(inject([Router], (router: Router) => {
+        router.relativeLinkResolution = 'legacy';
+
+        const fixture = createRoot(router, RootCmp);
+
+        router.navigateByUrl('/foo/bar');
+        advance(fixture);
+
+        const link = fixture.nativeElement.querySelector('a');
+        expect(link.getAttribute('href')).toEqual('/foo/bar/simple');
+      })));
+
+      it('should ignore empty paths in corrected mode', fakeAsync(inject([Router], (router: Router) => {
+        router.relativeLinkResolution = 'corrected';
+
+        const fixture = createRoot(router, RootCmp);
```

Comment:
```suggestion
        const fixture = createRoot(router, RelativeLinkCmp);
```

Reply:
The root component does not necessarily need to be RelativeLinkCmp, just the RootCmp should be good I think